### PR TITLE
Add Yang model support for adding Channel to PORT table

### PR DESCRIPTION
**Pull request by prgeor**
#### Why I did it

Add 'channel' to the CONFIG_DB PORT table. This will be needed to support PORT breakout to multiple channel ports so that Xcvrd can understand which datapath or channel to initialize on the CMIS compliant optics

#### How I did it
Add 'channel' to the CONFIG_DB PORT table.

#### How to verify it

Added unit test for valid and invalid channel number
Channel 0 -> No breakout
Channel 1 to 8 -> Breakout channel 1,2, ..8

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [X] 202205
- [X] 202211

#### Link to config_db schema for YANG module changes

[https://github.com/sonic-net/sonic-buildimage/blob/f011607a8a873b5b391aa54559a368998d25fd35/src/sonic-yang-models/doc/Configuration.md#port](https://github.com/sonic-net/sonic-buildimage/blob/f011607a8a873b5b391aa54559a368998d25fd35/src/sonic-yang-models/doc/Configuration.md#port)

### CODE REVIEW

**Comment by prgeor** [comment]
@shyam77git please review

**Comment by prgeor** [comment]
@ganglyu can you please review ?

File: src/sonic-yang-models/yang-models/sonic-port.yang [hunk 1]
```diff
@@ -121,6 +121,13 @@ module sonic-port{
...
 				}
 
+				leaf channel {
+					description "Logical channel(s) for physical port breakout";
+					type uint8 {
+						range 0..8;
+					}
+				}
+
 				leaf index {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-port.yang:131]: Any description?

**Review by ganglyu — APPROVED** [review]

**Comment by prgeor** [comment]
@lguohan can you help merge

**Comment by prgeor** [comment]
@yxieca please help merge to 202205

**Comment by prgeor** [comment]
@StormLiangMS please help merge to 202211

**Pull request merged**